src/dvclive/monitor_system.py

```diff
@@ -0,0 +1,265 @@
+                    f", but got '{disk_name}'."
+                )
+            disks_to_monitor[disk_name] = disk_path
+        return disks_to_monitor
+
+    def __call__(self, live):
+        self._live = live
+        self._shutdown_event = Event()
+        Thread(
+            target=self._monitoring_loop,
+        ).start()
+
+    def _monitoring_loop(self):
+        while not self._shutdown_event.is_set():
+            self._metrics = {}
+            for _ in range(self._num_samples):
+                try:
+                    last_metrics = self._get_metrics()
+                except psutil.Error:
+                    if self._warn_cpu_problem:
+                        logger.exception("Failed to monitor CPU metrics")
+                        self._warn_cpu_problem = False
+                except NVMLError:
+                    if self._warn_gpu_problem:
+                        logger.exception("Failed to monitor GPU metrics")
+                        self._warn_gpu_problem = False
+
+                self._metrics = merge_with(sum, self._metrics, last_metrics)
+                self._shutdown_event.wait(self._interval)
+                if self._shutdown_event.is_set():
+                    break
+            for name, values in self._metrics.items():
+                blacklisted = any(
+                    name.startswith(prefix) for prefix in self._plot_blacklist_prefix
+                )
+                self._live.log_metric(
+                    name,
+                    values / self._num_samples,
+                    timestamp=True,
+                    plot=None if blacklisted else self._plot,
+                )
+
+    def _get_metrics(self) -> Dict[str, Union[float, int]]:
+        result = {
+            **self._get_cpu_info(),
+            **self._get_ram_info(),
+            **self._get_disk_info(),
+        }
+        if GPU_AVAILABLE:
+            result.update(self.__get_gpu_info())
+        return result
+
+    def _get_ram_info(self) -> Dict[str, Union[float, int]]:
+        ram_info = psutil.virtual_memory()
+        return {
+            METRIC_RAM_USAGE_PERCENT: ram_info.percent,
+            METRIC_RAM_USAGE_GB: ram_info.used / GIGABYTES_DIVIDER,
+            METRIC_RAM_TOTAL_GB: ram_info.total / GIGABYTES_DIVIDER,
+        }
+
+    def _get_cpu_info(self) -> Dict[str, Union[float, int]]:
+        num_cpus = psutil.cpu_count()
+        cpus_percent = psutil.cpu_percent(percpu=True)
+        return {
+            METRIC_CPU_COUNT: num_cpus,
+            METRIC_CPU_USAGE_PERCENT: mean(cpus_percent),
+            METRIC_CPU_PARALLELIZATION_PERCENT: len(
+                [
+                    percent
+                    for percent in cpus_percent
+                    if percent >= MINIMUM_CPU_USAGE_TO_BE_ACTIVE
+                ]
+            )
+            * 100
+            / num_cpus,
+        }
+
+    def _get_disk_info(self) -> Dict[str, Union[float, int]]:
```

Comment:
If this is causing additional complications, I think we could drop it for now

Reply:
What complication do you see? :)

Reply:
I know there was discussion in the other PR like how to handle partitions. No problem if it's no longer an issue.